File: backend/app/Models/SchoolAssessment.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use MongoDB\Laravel\Eloquent\Model;

class SchoolAssessment extends Model
{
    use HasFactory;

    protected $collection = 'school_assessments';

    protected $fillable = [
        'child_profile_id',
        'subject',
        'assessment_type',
        'score',
        'max_score',
        'grade',
        'assessed_at',
        'teacher_note',
        'source',
    ];

    protected $casts = [
        'score' => 'float',
        'max_score' => 'float',
        'assessed_at' => 'datetime',
    ];

    protected $attributes = [
        'source' => 'manual',
    ];

    protected $appends = [
        'score_percent',
    ];

    public function childProfile()
    {
        return $this->belongsTo(ChildProfile::class, 'child_profile_id');
    }

    public function getScorePercentAttribute(): ?float
    {
        $score = $this->score;
        $maxScore = $this->max_score;

        if ($score === null || $maxScore === null || $maxScore <= 0) {
            return null;
        }

        return round(((float) $score / (float) $maxScore) * 100, 2);
    }
}

File: mobile/learny_app/lib/models/school_assessment.dart
class SchoolAssessment {
  const SchoolAssessment({
    required this.id,
    required this.subject,
    required this.assessmentType,
    required this.score,
    required this.maxScore,
    required this.assessedAt,
    this.grade,
    this.teacherNote,
    this.source = 'manual',
    this.scorePercent,
  });

  final String id;
  final String subject;
  final String assessmentType;
  final double score;
  final double maxScore;
  final DateTime assessedAt;
  final String? grade;
  final String? teacherNote;
  final String source;
  final double? scorePercent;

  factory SchoolAssessment.fromJson(Map<String, dynamic> json) {
    final id = (json['_id'] ?? json['id'] ?? '').toString();
    return SchoolAssessment(
      id: id,
      subject: json['subject']?.toString() ?? 'Unknown',
      assessmentType: json['assessment_type']?.toString() ?? 'unknown',
      score: (json['score'] as num?)?.toDouble() ?? 0,
      maxScore: (json['max_score'] as num?)?.toDouble() ?? 0,
      assessedAt:
          DateTime.tryParse(json['assessed_at']?.toString() ?? '') ??
          DateTime.fromMillisecondsSinceEpoch(0),
      grade: json['grade']?.toString(),
      teacherNote: json['teacher_note']?.toString(),
      source: json['source']?.toString() ?? 'manual',
      scorePercent: (json['score_percent'] as num?)?.toDouble(),
    );
  }
}
